python/cucim/src/cucim/skimage/morphology/convex_hull.py

```diff
@@ -0,0 +1,315 @@
+
+    # Find the convex hull
+    try:
+        hull = ConvexHull(coords)
+    except QhullError as err:
+        warn(
+            f"Failed to get convex hull image. "
+            f"Returning empty image, see error message below:\n"
+            f"{err}"
+        )
+        return cp.zeros(image.shape, dtype=bool)
+
+    coord_dtype = cp.min_scalar_type(max(image.shape))
+    if float64_computation:
+        float_dtype = cp.float64
+    else:
+        # float32 will be used if coord_dtype is <= 16-bit
+        # otherwise, use float64
+        float_dtype = cp.promote_types(cp.float32, coord_dtype)
+
+    kernel = get_coords_in_hull_kernel(coord_dtype, float_dtype, ndim)
+
+    if not image.flags.c_contiguous:
+        image = cp.ascontiguousarray(image)
+    convex_image = cp.empty_like(image)
+    hull_equations = cp.asarray(hull.equations, dtype=float_dtype)
+    kernel(image, hull_equations, tolerance, convex_image)
+    return convex_image.reshape(original_shape)
+
+
+def convex_hull_object(
+    image,
+    *,
+    connectivity=2,
+    float64_computation=False,
+    cpu_fallback_threshold=None,
+):
+    r"""Compute the convex hull image of individual objects in a binary image.
+
+    The convex hull is the set of pixels included in the smallest convex
+    polygon that surround all white pixels in the input image.
+
+    Parameters
+    ----------
+    image : (M, N) ndarray
+        Binary input image.
+    connectivity : {1, 2}, int, optional
+        Determines the neighbors of each pixel. Adjacent elements
+        within a squared distance of ``connectivity`` from pixel center
+        are considered neighbors.::
+
+            1-connectivity      2-connectivity
+                  [ ]           [ ]  [ ]  [ ]
+                   |               \  |  /
+             [ ]--[x]--[ ]      [ ]--[x]--[ ]
+                   |               /  |  \
+                  [ ]           [ ]  [ ]  [ ]
+
+    Extra Parameters
+    ----------------
+    float64_computation : bool, optional
+        If False, allow use of 32-bit float during the postprocessing stage
+    cpu_fallback_threshold : non-negative int or None
+        Number of pixels in an image before convex_hull_image will fallback
+        to pure CPU implementation.
+
+    Returns
+    -------
+    hull : ndarray of bool
+        Binary image with pixels inside convex hull set to ``True``.
+
+    Notes
+    -----
+    This function uses ``skimage.morphology.label`` to define unique objects,
+    finds the convex hull of each using ``convex_hull_image``, and combines
+    these regions with logical OR. Be aware the convex hulls of unconnected
+    objects may overlap in the result. If this is suspected, consider using
+    convex_hull_image separately on each object or adjust ``connectivity``.
+
+    The parameters listed under "Extra Parameters" above are present only
+    in cuCIM and not in scikit-image.
+    """
+    if connectivity not in tuple(range(1, image.ndim + 1)):
```

Comment:
There's no validation that the input is 2D or 3D. While the error will be caught by the connectivity check, it might be clearer to add an explicit dimensionality check:
```suggestion
    if image.ndim not in (2, 3):
        raise ValueError("Input image must be 2D or 3D")
    if connectivity not in tuple(range(1, image.ndim + 1)):
```